go/vt/tabletserver/streamlogger.go

```diff
@@ -150,50 +154,47 @@ func (stats *SQLQueryStats) FmtQuerySources() string {
 	return strings.Join(sources[:n], ",")
 }
 
-func (log *SQLQueryStats) RemoteAddr() string {
-	return callinfo.FromContext(log.context).RemoteAddr()
-}
-
-func (log *SQLQueryStats) Username() string {
-	return callinfo.FromContext(log.context).Username()
-}
-
-func (log *SQLQueryStats) ContextHTML() template.HTML {
-	return callinfo.FromContext(log.context).HTML()
-}
-
-func (log *SQLQueryStats) ErrorStr() string {
-	if log.Error != nil {
-		return log.Error.Error()
+// ErrorStr returns the error string or ""
+func (stats *SQLQueryStats) ErrorStr() string {
+	if stats.Error != nil {
+		return stats.Error.Error()
 	}
 	return ""
 }
 
-// String returns a tab separated list of logged fields.
-func (log *SQLQueryStats) Format(params url.Values) string {
+// Format returns a tab separated list of logged fields.
+func (stats *SQLQueryStats) Format(params url.Values) string {
 	_, fullBindParams := params["full"]
+
+	remoteAddr := ""
+	username := ""
+	ci, ok := callinfo.FromContext(stats.context)
+	if ok {
+		remoteAddr = ci.RemoteAddr
+		username = ci.Username
+	}
 	return fmt.Sprintf(
 		"%v\t%v\t%v\t%v\t%v\t%.6f\t%v\t%q\t%v\t%v\t%q\t%v\t%.6f\t%.6f\t%v\t%v\t%v\t%v\t%v\t%v\t%q\t\n",
-		log.Method,
-		log.RemoteAddr(),
-		log.Username(),
-		log.StartTime.Format(time.StampMicro),
-		log.EndTime.Format(time.StampMicro),
-		log.TotalTime().Seconds(),
-		log.PlanType,
-		log.OriginalSql,
-		log.FmtBindVariables(fullBindParams),
-		log.NumberOfQueries,
-		log.RewrittenSql(),
-		log.FmtQuerySources(),
-		log.MysqlResponseTime.Seconds(),
-		log.WaitingForConnection.Seconds(),
-		log.RowsAffected,
-		log.SizeOfResponse(),
-		log.CacheHits,
-		log.CacheMisses,
-		log.CacheAbsent,
-		log.CacheInvalidations,
-		log.ErrorStr(),
+		stats.Method,
+		remoteAddr,
```

Comment:
This may break some backward compatibility. I checked that querylogz.go doesn't use this, but others may.